mlflow/tensorflow/__init__.py

```diff
@@ -734,22 +793,28 @@ def predict(self, data):
 
 
 class _KerasModelWrapper:
-    def __init__(self, keras_model, graph, sess):
+    def __init__(self, keras_model, signature):
         self.keras_model = keras_model
-        self._graph = graph
-        self._sess = sess
+        self.signature = signature
 
     def predict(self, data):
-        def _predict(data):
-            if isinstance(data, pandas.DataFrame):
-                predicted = pandas.DataFrame(self.keras_model.predict(data.values))
-                predicted.index = data.index
```

Comment:
This is a breaking change right? Don't we need to wrap the output in a pandas DataFrame if the input is a pandas DataFrame? Can we add a test for this.

Reply:
I prefer to do not wrap result as dataframe because:

In mlflow master code, if we specified model signature and input is pandas dataframe, then keras pyfunc model prediction results will still be numpy array (in _enforce_signature the dataframe is converted as dict and passed to keras model as input)

for other flavors like sklearn / pytorch , if input is dataframe, output is also numpy array

Only the case (keras model, no signature and input is dataframe), the output is dataframe. Not sure why previously we do this

Reply:
Got it - we can't make a breaking change, unfortunately, so we need to return dataframe